### ISSUE TITLE
Rename note under user hierarchy breaks expansion links 

### ISSUE THREAD

**Issue by nickolay-kondratyev** [dendronhq/dendron#3780]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [ ] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features
- [x] Refactor

## Describe the bug

Renaming notes that are under `user.` hierarchy leads to breaking expansion links.

## To Reproduce

Create note:

- `user.Tester.th.HelloWorld` 
- `note1` 

In note1 add

```
![[user.Tester.th.HelloWorld]]
```

Use `Rename note` on `user.Tester.th.HelloWorld` to rename it to `user.Tester.th.HelloWorld2`.

Expected behavior note1 contents are expected to become:
```
![[user.Tester.th.HelloWorld2]]
```

Actual behavior note1 contents become:
```
@Tester.th.HelloWorld2
```

Expansion is broken.

## Desktop (please complete the following information)

- Browser (if with published websites; include version): [e.g. chrome, safari]
- Versions (all that apply to bug):
  - Dendron Extension: 117
  - VS Code version:
  - Dendron CLI:

**Comment by Harshita-mindfire** [comment]
Thanks @nickolay-kondratyev for reporting the bug. I can repro this issue. We will ping here when this gets fixed 🙏🏻

**Comment by Harshita-mindfire** [comment]
This should be fixed after Dendron version 0.118.0. Feel free to comment here if the issue still persists.
